### PR TITLE
feat: add Kubernetes intake module and kubernetes-audit-log intake format

### PR DESCRIPTION
**Pull request by maxime-el-haddari**
This PR is to add publicly a custom intake done for SEKOIA.IO and monitor our Kubernetes cluster.
Thanks for your review since I have some doubts on some syntax aspects.

### CODE REVIEW

File: Kubernetes/audit-log/_meta/fields.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  name: action.properties.KubernetesNamespace
+  type: keyword
+  short: action.properties.KubernetesNamespace
+  description: "kubernetes namespace"
+
+action.properties.KubernetesName:
+  name: action.properties.KubernetesName
+  type: keyword
+  short: action.properties.KubernetesName
+  description: "kubernetes object name"
```
Comment (squioc) [inline Kubernetes/audit-log/_meta/fields.yml:19]:
```suggestion
kubernetes.object.name:
  name: kubernetes.object.name
  type: keyword
  short: kubernetes.object.name
  description: "The name of the object"
```

File: Kubernetes/audit-log/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  name: action.properties.KubernetesRBACDecision
+  type: keyword
+  short: action.properties.KubernetesRBACDecision
+  description: "RBAC decision for the kubernetes action"
+
+action.properties.KubernetesKind:
+  name: action.properties.KubernetesKind
+  type: keyword
+  short: action.properties.KubernetesKind
+  description: "Kind of the kubernetes resource"
```
Comment (squioc) [inline Kubernetes/audit-log/_meta/fields.yml:31]:
```suggestion
kubernetes.kind:
  name: kubernetes.kind
  type: keyword
  short: kubernetes.kind
  description: "The nature of the resource"
```

File: Kubernetes/audit-log/_meta/fields.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  name: action.properties.KubernetesName
+  type: keyword
+  short: action.properties.KubernetesName
+  description: "kubernetes object name"
+
+action.properties.KubernetesRBACDecision:
+  name: action.properties.KubernetesRBACDecision
+  type: keyword
+  short: action.properties.KubernetesRBACDecision
+  description: "RBAC decision for the kubernetes action"
```
Comment (squioc) [inline Kubernetes/audit-log/_meta/fields.yml:25]:
```suggestion
kubernetes.rbac.decision:
  name: kubernetes.rbac.decision
  type: keyword
  short: kubernetes.rbac.decision
  description: "RBAC decision for the action"
```

File: Kubernetes/audit-log/_meta/fields.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  name: action.properties.KubernetesKind
+  type: keyword
+  short: action.properties.KubernetesKind
+  description: "Kind of the kubernetes resource"
+
+action.properties.KubernetesSubresource:
+  name: action.properties.KubernetesSubresource
+  type: keyword
+  short: action.properties.KubernetesSubresource
+  description: "Subresource kubernetes"
```
Comment (squioc) [inline Kubernetes/audit-log/_meta/fields.yml:37]:
```suggestion
kubernetes.subresource:
  name: kubernetes.subresource
  type: keyword
  short: kubernetes.subresource
  description: "The subresource"
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for the contribution.
For custom fields, we highly recommends prefixing with the vendor name (cf: [Taxonomy](https://github.com/SEKOIA-IO/intake-formats/blob/main/doc/taxonomy.md#custom-field))

File: Kubernetes/_meta/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+uuid: b633239c-2c91-4c1d-9da6-c361638a9064
+name: Kubernetes
+slug: kubernetes
+description: 'Kubernetes is an open-source container orchestration system for automating software deployment, scaling and managment.'
```
Comment (mortrevere) [inline Kubernetes/_meta/manifest.yml:4]:
```suggestion
description: 'Kubernetes is an open-source container orchestration system for automating software deployment, scaling and management.'
```

File: Kubernetes/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Kubernetes (also known as K8s) is an open-source container orchestration system for automating software deployment, scaling and management.
```
Comment (mortrevere) [inline Kubernetes/README.md:1]:
```suggestion
Kubernetes (also known as k8s) is an open-source container orchestration system for automating software deployment, scaling and management.
```

**Review by squioc — DISMISSED** [review]

File: Kubernetes/audit-log/_meta/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+uuid: 4a3bb630-951a-40d9-be5e-5c712b37248e
+name: Kubernetes Audit Log
+slug: kubernetes-audit-log
+description: >-
```
Comment (restanrm) [inline Kubernetes/audit-log/_meta/manifest.yml:4]:
```suggestion
data_sources: 
  Kubernetes Audit Log: Audit logs of Kubernetes clusters
description: >-
```

Comment (restanrm) [reply Kubernetes/audit-log/_meta/manifest.yml:4]: I've suggested to add data source to be able to link rules to this source.

File: Kubernetes/audit-log/_meta/manifest.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+uuid: 4a3bb630-951a-40d9-be5e-5c712b37248e
+name: Kubernetes Audit Log
+slug: kubernetes-audit-log
+data_sources: 
+  Kubernetes Audit Log: Audit logs of Kubernetes clusters
```
Comment (restanrm) [inline Kubernetes/audit-log/_meta/manifest.yml:5]:
```suggestion
  Kubernetes Audit Logs: Audit logs of Kubernetes clusters
```

File: Kubernetes/audit-log/_meta/smart-descriptions.json [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+                "field": "kubernetes.resource"
+            },
+            {
+                "field": "kubernetes.object.name"
+            },
+            {
+                "field": "kubernetes.namespace"
+            }
+        ],
+        "relationships": []
```
Comment (mortrevere) [inline Kubernetes/audit-log/_meta/smart-descriptions.json:26]: Most relationships in this file are empty, I think they could be filled easily (like line 71 is)

**Review by mortrevere — CHANGES_REQUESTED** [review]

File: Kubernetes/audit-log/tests/event_user_exec_in_pod.json [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+      "object": {
+        "name": "test-1669140000-zp58r"
+        },
+      "namespace": "test",
+      "subresource": "exec",
+      "resource": "pods",
+      "rbacreason": "RBAC: allowed by ClusterRoleBinding test-role-binding of ClusterRole test-admin to Group system:authenticated"
+    },
+    "user": {
+      "name": "user@mail.com",
```
Comment (squioc) [inline Kubernetes/audit-log/tests/event_user_exec_in_pod.json:24]: Look like an email. In this case, `user.email` is recommended.

**Review by squioc — APPROVED** [review]

File: Kubernetes/audit-log/tests/event_user_exec_in_pod.json (hunk 10 above)
Comment (maxime-el-haddari) [reply Kubernetes/audit-log/tests/event_user_exec_in_pod.json:24]: @squioc In this case, it is an email as it is often done but it is not enforced as an email and k8s could use any string as a username such as `kube-admin` for example.

**Pull request merged**